os/drivers/input/touchscreen.c

```diff
@@ -232,17 +238,18 @@ static int touch_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
 	}
 
 	touch_semtake(&priv->sem, false);
-
 	switch (cmd) {		
 #if defined(CONFIG_TOUCH_CALLBACK)
 		case TSIOC_SETAPPNOTIFY: {
 			struct touch_set_callback_s *touch_app = (struct touch_set_callback_s *)arg;
 			priv->app_touch_point_buffer = touch_app->touch_points;
 			priv->is_touch_detected = touch_app->is_touch_detected;
 			touchvdbg("App notification callback register is successful\n");
+			priv->ops->touch_enable(priv);
 		}
 		break;
 #endif
+#ifdef CONFIG_TOUCH_CALLBACK
```

Comment:
Why is TSIOC_DISABLE and enable inside touch_Callback condition only

Reply:
The apps is already having controllability, the app can don't read touch.
So if CONFIG_TOUCH_CALLBACK is disable, we don't need to provide enable and disable